### PR TITLE
68 triggers of services to the api

### LINKED ISSUE #68: Triggers of services to the API
**Short description:**
Add available triggers of services on API

**Describe the solution you'd like**
API that sends triggers on services (like all available servers/channels on Discord, available boards on Trello etc...).

### PR DESCRIPTION
**Pull request by tbellicha**
# Purpose

The purpose of this pull request is to add Discord getters functions and to implement them in the API

Fixes #68

## Added features

* It adds several routes:
  *api/services/discord/getTextChannels
  *api/services/discord/getVoiceChannels
  *api/services/discord/getAvailableGuilds

## Updated features

* Added `isRequired` field in Reaction in the Database, because not all field will be required.
* Changed `parameters` to `Parameters` in Reaction as well, to fit the new format.

### CODE REVIEW

File: server/server_app.js [hunk 1]
```diff
@@ -412,6 +415,42 @@ app.get(
   }
 )
 
+/**
+ * @brief List all available Voice Channels on a given GuildID.
+ */
```
Comment (LucasTesnier) [inline server/server_app.js:421]: Need to be a get

File: server/server_app.js [hunk 2]
```diff
@@ -412,6 +415,42 @@ app.get(
...
+    status: 'success',
+    data: channels,
+    statusCode: res.statusCode
+  })
+})
+
+/**
+ * @brief List all available Text Channels on a given GuildID.
+ */
+app.post('/api/services/discord/getTextChannels', async (req, res) => {
```
Comment (LucasTesnier) [inline server/server_app.js:435]: Same

File: server/server_app.js [hunk 3]
```diff
@@ -412,6 +415,42 @@ app.get(
...
+    status: 'success',
+    data: channels,
+    statusCode: res.statusCode
+  })
+})
+
+/**
+ * @brief List all available Guilds where the bot is.
+ */
+app.post('/api/services/discord/getAvailableGuilds', async (req, res) => {
```
Comment (LucasTesnier) [inline server/server_app.js]: Same

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Just change your useless post to get otherwise NTR

**Review by LucasTesnier — DISMISSED** [review]
AMAZING

**Review by ZQUEMA — CHANGES_REQUESTED** [review]
You have to change the function at line 551 in server/server_app.js because the creation of a parameter has changed

**Review by TomDUVAL-MAHE — DISMISSED** [review]
NTR

**Review by ZQUEMA — APPROVED** [review]

**Review by GuyomT — APPROVED** [review]
NTR

**Pull request merged**